### ISSUE TITLE
ng-model not populated by files on change

### ISSUE THREAD

**Issue by Sebi55** [danialfarid/ng-file-upload#986]
Hey thanks for the great plugin.
I have got the problem that I want the user to select an image, he can than see a preview of it and then hit the submit button to really upload it. I already got another upload working where I upload a video to my server but I somehow cannot get the image upload to work. When I drop the image it works perfectly and I can see the preview but if select it by clicking instead of dragging the model is not updated, if try to use ngf-change and manually place the file into files I get the error that file is null and if I do $watch then I also get back that it is null. Furthermore if I select a file for the first time nothing happens it only does anything on the second try to select something.
The crazy thing is that it works for me for the video without any problem. The only difference is that I directly upload the video instead of waiting for a submit but that should not change the select thing shouldn't it?
Here my code:
```html
<form name="create-artist">
    	<input type="text" name="artistname" placeholder="Band- oder Künstlername" />
      	<div ngf-drop ngf-select ng-model="files" ngf-multiple="true" ngf-pattern="image/*" accept="image/*" ngf-validate="{pattern: 'image/*'}" class="upload-container">
			<div class="content">
				<h5>Lade Dein Video Hoch</h5>
				<h6>Videodatei für Upload ziehen und ablegen</h6>
				<div class="upload-icon"></div>
				<span class="t">Oder:</span>
				<div ngf-pattern="image/*" accept="image/*" ngf-validate="{pattern: 'image/*'}">
					<span>Durchsuchen</span>
				</div>
				<h6>Dein Video wird dann sofort hochgeladen!</h6>
			</div>
			<img ngf-src="files[0]">
		</div>
      <input type="button" name="create" value="Erstellen" ng-click="doit()"/>
 </form>
```
And here my controller:
```javascript
app.controller('createArtistCtrl', function($scope, Login, $location, Upload, $interval){
    $scope.$watch('files', function () {
        console.log($scope.files);
    });
    $scope.upload = function(files) {
        $scope.progressPercentage = 0;
        $scope.uploaded = false;
        var video_data = {
            "title": $scope.title,
            "genre_id": $scope.genre
        };
        console.log(video_data);
        if ($scope.files.length > 0) {
            Upload.upload({
                url: base + '/postImage',
                headers: {
                    'Content-Type': $scope.files[0].type
                },
                method: 'POST',
                data: $scope.files[0],
                file: $scope.files[0],
            }).progress(function(evt) {
                $scope.progressPercentage = parseInt(100.0 * evt.loaded / evt.total);
            }).success(function(data, status, headers, config) {
                console.log(data);
                $scope.path = data;
                $scope.uploaded = true;
            });
        }
    };
   $scope.doit= function(){
   	console.log($scope.files);
   }
});
```
Maybe I should mention that this form is shown inside a ngDialog modal. I apply the controller using like that: 
```javascript
ngDialog.open({
    		template: '../../themes/default/createartistmodal.html',
    		scope: $scope,
    		controller: 'createArtistCtrl'
    	});
```

It should not be a problem since a new controller is applied and anything is done inside this controller and the drop function is working so there need to be another error.
I hope you can help me.

**Comment by danialfarid** [comment]
What version of the plugin?

**Comment by Sebi55** [comment]
We are using 7.0.15

**Comment by danialfarid** [comment]
You can upgrade to 7.0.16 but I doubt it would fix anything.
Looking at your code everything looks fine. If you could create a jsfiddle I will debug it and let you know where the issue is.

**Comment by Sebi55** [comment]
I updated to the newest version but no change.
I quickly created a fiddle.
I tested it my self and it has got something to do with the modal maybe you can find out.
The modal opens twice if you hit the button again so you should not do that.
You will see that you can drop the image and it will work without any problem but when you select it nothing happens.
http://jsfiddle.net/sm1q4m32/3/

**Comment by danialfarid** [comment]
It was a bug, fixed in version 7.0.17, thanks for reporting.

**Comment by Sebi55** [comment]
Thanks works great.
